### DISCUSSION TITLE
Code block - syntax and block highlighting

### DISCUSSION THREAD

**Discussion by KlapenHz** [nvim-neorg/neorg#1135] [Q&A]
Hi, 
1. Is there a way to to somehow distinguish a code block? E.g. by changing the background?
2. Does the @code language require any additional parameters to change the colors of the syntax in the block.
Thanks for any suggestions.

**Comment by Anrock** [comment]
1. Do you mean like having different colors for different code blocks? Because now everything between `@code` and `@end` should already have different background and syntax highlight.
2. I don't think `@code` accepts anything to customize color scheme. The only parameter there is a language name to use specific treesitter grammar for higlighting.

**Reply by KlapenHz** [reply]
This is my example of test index file:
![image](https://github.com/nvim-neorg/neorg/assets/110558378/bc59c3f3-b19d-4db9-a994-a38a8f735805)
and it doesn't have any different colors inside. Maybe it's lack of some configuration on my site?

**Reply by Anrock** [reply]
More like colorscheme issue where code background color is too samey to usual background to notice + transparency + image on the background.
Try `:colorscheme delek` and turn off transparency and see if it makes any difference
![2023-11-01-175643_442x415_scrot](https://github.com/nvim-neorg/neorg/assets/1689392/7533843a-7a69-4e1d-a751-5e315e925a89)

**Reply by KlapenHz** [reply]
Nothing changed:
![image](https://github.com/nvim-neorg/neorg/assets/110558378/1a362ba7-c269-4931-9548-60d1733aaeb9)
I think in this case, some keywords should at least be recolored, but unfortunately everything is the same here.
I also removed color scheme and all config from my terminal and it didn't help:
![image](https://github.com/nvim-neorg/neorg/assets/110558378/5f3bc771-a7dc-410f-a35e-b60468b8798b)
![image](https://github.com/nvim-neorg/neorg/assets/110558378/a0b24f52-a0e4-447c-8867-531b8c2ad0a7)
Also I tried to run it in another terminal, and also problem exists.
![image](https://github.com/nvim-neorg/neorg/assets/110558378/a75050b8-6503-43e2-ad7c-37c298e66544)

**Reply by katawful** [reply]
you need to have python and bash treesitter parsers installed for syntax highlighting to work out of the box. neither are supplied for 0.9.x (both are supplied with 0.10-devel releases). for languages that don't have a parser yet, you need to enable `core.syntax` in your neorg config, but bash and python have parsers

**Reply by KlapenHz** [reply]
Thanks @katawful, I didn't have parser. Now syntax has colors. 
And how make background in different color? Like this below:
![image](https://github.com/nvim-neorg/neorg/assets/110558378/5a32f997-727c-420d-8d01-16fcd4d74373)
